Tests/AppTests/SearchTests.swift

```diff
@@ -448,9 +435,44 @@ class SearchTests: AppTestCase {
         XCTAssertEqual(res, .init(hasMoreResults: false, results: []))
     }
 
-    func renderSQL(_ query: SQLExpression?) -> String {
-        var serializer = SQLSerializer(database: app.db as! SQLDatabase)
-        query?.serialize(to: &serializer)
-        return serializer.sql
+    func test_search_topic() throws {
+        // Test searching for a topic
+        // setup
+        // p1: decoy
+        // p2: match
+        let p1 = Package(id: .id1, url: "1", score: 10)
+        let p2 = Package(id: .id2, url: "2", score: 20)
+        try [p1, p2].save(on: app.db).wait()
+        try Repository(package: p1,
+                       defaultBranch: "main",
+                       name: "1",
+                       owner: "foo",
+                       summary: "").save(on: app.db).wait()
+        try Repository(package: p2,
+                       defaultBranch: "main",
+                       keywords: ["topic"],
+                       name: "2",
+                       owner: "foo",
+                       summary: "").save(on: app.db).wait()
+        try Version(package: p1, packageName: "p1", reference: .branch("main"))
+            .save(on: app.db).wait()
+        try Version(package: p2, packageName: "p2", reference: .branch("main"))
+            .save(on: app.db).wait()
+        try Search.refresh(on: app.db).wait()
+
+        // MUT
+        let res = try Search.fetch(app.db, ["topic"], page: 1, pageSize: 20).wait()
+
+        XCTAssertEqual(res.results, [
+            .keyword(.init(keyword: "topic")),
+            // TODO: also match against package keywords
+            //            .package(.init(packageId: .id2,
+            //                           packageName: "2",
+            //                           packageURL: "2",
+            //                           repositoryName: "2",
+            //                           repositoryOwner: "foo",
+            //                           summary: ""))
```

Comment:
This is the other piece that I've skipped in order to get the first version out: matching against a package's keywords.

One simple way to do this is by appending the topics to the search string:

```sql
WHERE CONCAT_WS(' ', "package_name", COALESCE("summary", ''), "name", "owner", ARRAY_TO_STRING(keywords, ' ')) ~* 'foo'
```

I _think_ we'll want to do that. In practise it might pull in a lot of results for certain terms, like for instance `test`, but I suppose that's a good thing. You can sort of do a keyword search without actually picking the keyword result from the search results.